TeachingRecordSystem/src/TeachingRecordSystem.Core/Services/TrsDataSync/TrsDataSyncHelper.cs

```diff
@@ -89,6 +90,78 @@ public static (string EntityLogicalName, string[] AttributeNames) GetEntityInfoF
         };
     }
 
+    private static InductionInfo? MapInductionInfoFromDqtInduction(
+        dfeta_induction? induction,
+        Contact contact,
+        bool ignoreInvalid)
+    {
+        // Double check that contact record induction status matches the induction record (if there is one) induction status (which should have been set via CRM plugin)
+        var hasQtls = contact.dfeta_QtlsDateHasBeenSet == true;
+        if (induction is not null && induction.dfeta_InductionStatus != contact.dfeta_InductionStatus)
+        {
+            if (ignoreInvalid)
+            {
+                return null;
+            }
+
+            throw new InvalidOperationException($"Induction status for contact {contact.ContactId} does not match induction status for induction {induction!.dfeta_inductionId}.");
```

Comment:
```suggestion
            throw new InvalidOperationException($"Induction status for contact {contact.ContactId} does not match induction status for induction {induction.dfeta_inductionId}.");
```

Reply:
Might be worth adding both statuses to the exception message here so it's super clear in any error logs.